src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy

```diff
@@ -48,29 +48,42 @@ abstract class AbstractDockerRemoteApiTask extends DefaultTask {
     @Input
     @Optional
     String apiVersion
-    
+
     /**
      * Closure to handle the possibly throw exception
      */
     @Input
     @Optional
     Closure onError
 
+    @Input
+    @Optional
+    Closure onNext
+
+    @Input
+    @Optional
+    Closure onComplete
```

Comment:
@orzeh the more I think about this the more I think it makes more sense to create an `AbstractReactiveStreamsTask`. This class will extend that and the `AbstractReactiveStreamsTask` will extend DefaultTask. I think, from a documentation perspective, it will be easier to talk at length about these tasks, and how we implement them, vs where things are now and it being intertwined with our current docs. 

Don't think is over-engineering as we are just adding 1 more class. But this will give us a single place to point devs to for information on how and/or why we are implementing these methods.